flight/PiOS/STM32F10x/pios_servo.c

```diff
@@ -239,44 +293,30 @@ void PIOS_Servo_OneShot_Set(uint8_t servo, uint16_t position)
 			break;
 	}
 }
+#endif /* PIOS_INCLUDE_HPWM */
 
+#if defined(PIOS_INCLUDE_HPWM)
 /**
-* Update the timer for OneShot
+* Update the timer for HPWM/OneShot
 */
-void PIOS_Servo_OneShot_Update()
+void PIOS_Servo_Update()
 {
 	if (!servo_cfg) {
 		return;
 	}
 
-	TIM_TimeBaseInitTypeDef TIM_TimeBaseStructure;
-	TIM_TimeBaseStructInit(&TIM_TimeBaseStructure);
-
 	for (uint8_t i = 0; i < servo_cfg->num_channels; i++) {
 		const struct pios_tim_channel * chan = &servo_cfg->channels[i];
 
-		/* Look for a disabled timer which is probably used by OneShot */
-		if (!(chan->timer->CR1 & TIM_CR1_CEN)) {
-			/* Choose the correct prescaler value for the APB the timer is attached */
-			if (chan->timer==TIM6 || chan->timer==TIM7) {
-				// These timers cannot be used here.
-				continue;
-			} else if (chan->timer==TIM1 || chan->timer==TIM8) {
-				if (PIOS_PERIPHERAL_APB2_CLOCK == PIOS_SYSCLK)
-					TIM_TimeBaseStructure.TIM_Prescaler = (PIOS_PERIPHERAL_APB2_CLOCK / OneShotFrequency) - 1;
-				else
-					TIM_TimeBaseStructure.TIM_Prescaler = (PIOS_PERIPHERAL_APB2_CLOCK*2 / OneShotFrequency) - 1;
-			} else {
-				if (PIOS_PERIPHERAL_APB1_CLOCK == PIOS_SYSCLK)
-					TIM_TimeBaseStructure.TIM_Prescaler = (PIOS_PERIPHERAL_APB1_CLOCK / OneShotFrequency) - 1;
-				else
-					TIM_TimeBaseStructure.TIM_Prescaler = (PIOS_PERIPHERAL_APB1_CLOCK*2 / OneShotFrequency) - 1;
-			}
-
+		/* Check for channels that are using synchronous output */
+		/* Look for a disabled timer using synchronous output */
+		if (!(chan->timer->CR1 & TIM_CR1_CEN) &&
+		     (output_channel_mode[i] == SYNC_PWM_TRUE)) {
 			/* enable it again and reinitialize it */
 			TIM_Cmd(chan->timer, ENABLE);
```

Comment:
@ernieift when you say "enabling the timer" do you mean move this line outside the conditional? Before or after?